src/camera/arc_ball.rs

```diff
@@ -439,18 +422,20 @@ impl Camera for ArcBall {
     fn update(&mut self, _: &Canvas) {}
 }
 
-#[derive(Clone, Copy, PartialEq, Eq, Debug)]
-enum CoordSystem {
-    RightHandYUp,
-    RightHandZUp,
+#[derive(Clone, Copy, Debug)]
+struct CoordSystemRh {
+    up_axis: Unit<Vector3<f32>>,
+    rotation_to_y_up: UnitQuaternion<f32>,
 }
 
-impl CoordSystem {
+impl CoordSystemRh {
     #[inline]
-    fn up_axis(self) -> Vector3<f32> {
-        match self {
-            CoordSystem::RightHandYUp => Vector3::y(),
-            CoordSystem::RightHandZUp => Vector3::z(),
+    fn from_up_axis(up_axis: Unit<Vector3<f32>>) -> Self {
+        let rotation_to_y_up =
+            UnitQuaternion::rotation_between_axis(&up_axis, &Vector3::y_axis()).unwrap();
```

Comment:
The `unwrap` will panic if `up_axis` is set to `-Y`. So this should probably by something like `.or_else(|| UnitQuaternion::new(Vector3::x() * f32::PI))` (needs to be tested).

Reply:
Sorry, I messed this one up! Should be fixed.